src/apme_engine/daemon/launcher.py

```
@@ -112,17 +246,20 @@ def _health_check(address: str, timeout: float = 3.0) -> bool:
+    from apme.v1 import engine_pb2_grpc
     from apme.v1.common_pb2 import HealthRequest
 
+    channel = None
     try:
         channel = grpc.insecure_channel(address)
-        stub = primary_pb2_grpc.PrimaryStub(channel)  # type: ignore[no-untyped-call]
+        stub = engine_pb2_grpc.EngineStub(channel)  # type: ignore[no-untyped-call]
         resp = stub.Health(HealthRequest(), timeout=timeout)
-        channel.close()
         return bool(resp.status == "ok")
     except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)